### PR TITLE
 added config creation for GameServerApp

### PR DESCRIPTION
**Pull request by LeagueRaINi**
wasnt sure about the GameServerConfig so i didnt touch it for now

### CODE REVIEW

File: GameServerApp/Program.cs [hunk 1]
```diff
@@ -84,17 +139,17 @@ private static void Main(string[] args)
 
     public class ArgsOptions
     {
-        [Option("config", DefaultValue = "Settings/GameInfo.json")]
-        public string ConfigPath { get; set; }
+        [Option("config", DefaultValue = "Settings\\GameInfo.json")]
```
Comment (MythicManiac) [inline GameServerApp/Program.cs]: Use linux paths (unless they don't work on Windows, I believe they do in this case)

File: GameServerApp/Program.cs [hunk 2]
```diff
@@ -84,17 +139,17 @@ private static void Main(string[] args)
...
+        public string GameInfoJsonPath { get; set; }
 
-        [Option("config-gameserver", DefaultValue = "Settings/GameServerSettings.json")]
-        public string ConfigGameServerPath { get; set; }
+        [Option("config-gameserver", DefaultValue = "Settings\\GameServerSettings.json")]
```
Comment (MythicManiac) [inline GameServerApp/Program.cs]: Use linux paths (unless they don't work on Windows, I believe they do in this case)

File: GameServerApp/Program.cs [hunk 3]
```diff
@@ -15,44 +17,97 @@ internal class Program
...
+
+            var parsedArgs = ArgsOptions.Parse(args);
+            var settingsPath = Path.Combine(Directory.GetCurrentDirectory(), "Settings");
 
-            var configJson = options.ConfigJson;
-            if (string.IsNullOrEmpty(configJson))
+            if (string.IsNullOrEmpty(parsedArgs.GameInfoJson))
             {
-                configJson = File.ReadAllText(options.ConfigPath);
+                if (File.Exists(parsedArgs.GameInfoJsonPath))
```
Comment (MythicManiac) [inline GameServerApp/Program.cs]: I'd flip this `if .. else` block the other way around, doing:
```CS
if(!File.Exists(parsedArgs.GameInfoJsonPath))
{
    // generate folders and file to the target path
}
// load settings file from target path
```

File: GameServerApp/Program.cs [hunk 4]
```diff
@@ -15,44 +17,97 @@ internal class Program
...
+                {
+                    parsedArgs.GameInfoJson = File.ReadAllText(parsedArgs.GameInfoJsonPath);
+                }
+                else
+                {
+                    try
+                    {
+                        Directory.CreateDirectory(settingsPath);
```
Comment (MythicManiac) [inline GameServerApp/Program.cs]: what's up with the odd spacing?

**Review by MythicManiac — COMMENTED** [review]
Did you forget to include the actual resources? I don't see any binary blob being added at least, nor a replacement for the removed json files.

Personally I'm not really a fan of the resources system if it ruins version management for the default configs. If those can be included to the resources as text files that exist on the solution but are built inside the .exe, then it's fine.

**Comment by MythicManiac** [comment]
Nice work though, autogenerating default config is definitely welcome 👍

**Comment by LeagueRaINi** [comment]
it uses the files in "LeagueSandbox\GameServer\GameServerApp\Settings" had problems with ".template" so i had to remove it

File: GameServerApp/Program.cs (hunk 4 above)
Comment (LeagueRaINi) [reply GameServerApp/Program.cs]: its the normal spacing vs applies

Comment (MythicManiac) [reply GameServerApp/Program.cs]: I mean the newlines under this?

**Comment by MythicManiac** [comment]
It should use the template, as that's included in the repository, and whatever it uses should be coming from the repository

File: GameServerApp/Program.cs (hunk 4 above)
Comment (LeagueRaINi) [reply GameServerApp/Program.cs]: oh hm i just thinks its more readable then writing everything under eachother without a single space

**Comment by MythicManiac** [comment]
I think you misunderstood a bit what I meant about the `if ... else` block, my bad for not being clear enough.

Consider the following:
```CS
        private static void LoadConfigOrCreateDefault(path, currentValue, defaultValue)
        {
            if (!string.IsNullOrEmptu(currentValue)) {
                return currentValue;
            }
            if (!File.Exists(path))
            {
                Directory.CreateDirectory(Path.GetDirectoryName(path));
                File.WriteAllText(
                    path,
                    Encoding.UTF8.GetString(defaultValue)
                );
            }
            return File.ReadAllText(parsedArgs.GameInfoJsonPath);
        }

        private static void Main(string[] args)
        {
            _logger = LoggerProvider.GetLogger();

            var parsedArgs = ArgsOptions.Parse(args);
            var settingsPath = Path.Combine(Directory.GetCurrentDirectory(), "Settings");

            parsedArgs.GameInfoJson = LoadConfigOrCreateDefault(
                parsedArgs.GameInfoJsonPath,
                parsedArgs.GameInfoJson,
                Encoding.UTF8.GetString(Resources.GameInfo)
            );
            configGameServerSettings = GameServerConfig.LoadFromJson(LoadConfigOrCreateDefault(
                parsedArgs.GameServerSettingsJsonPath,
                parsedArgs.GameServerSettingsJson,
                Encoding.UTF8.GetString(Resources.GameServerSettings)
            ))

            var gameServerBlowFish = "17BLOhi6KZsTtldTsizvHg==";
            var gameServerLauncher = new GameServerLauncher(parsedArgs.ServerPort, parsedArgs.GameInfoJson, gameServerBlowFish);

...
```

Less copypaste code

**Comment by LeagueRaINi** [comment]
aight will look into that after i have slept

**Review by MythicManiac — APPROVED** [review]
Nice work 👍

**Pull request merged**